subsys/event_manager/event_manager_priv.h

```diff
@@ -184,22 +173,8 @@ extern "C" {
 	const struct event_type _CONCAT(__event_type_, ename) __used _EM_FORCED_ALIGNMENT	\
 	__attribute__((__section__("event_types"))) = {						\
```

Comment:
can't we use `STRUCT_SECTION_ITERABLE` for `event_types` array?

Reply:
There are two problems with that: 
1. There is problem with accessing section start address, sth like &__start_event_types is not seen from c file and throws erro undefined __start_event_types variable.
2. Do not know how to ad keyword using  STRUCT_SECTION_ITERABLE. And if I do not use them I get "conflicting types" error.

Reply:
section types do not need tags in them so we don't need to place empty structures

I have pushed the change in the review commit